influx/annotations.go

```diff
@@ -220,33 +244,80 @@ func (r *influxResults) Annotations() (res []chronograf.Annotation, err error) {
 	annos := map[string]annotationResult{}
 	for _, u := range *r {
 		for _, s := range u.Series {
+			columnIndex := map[string]int{}
+
+			for i, c := range s.Columns {
+				columnIndex[c] = i
+			}
+
+			indexOf := func(k string) (int, error) {
+				i, prs := columnIndex[k]
+
+				if !prs {
+					return -1, fmt.Errorf("Could not find %q in annotation", k)
+				}
+
+				return i, nil
+			}
+
 			for _, v := range s.Values {
 				anno := annotationResult{}
+				var i int
 
-				if anno.EndTime, err = v.Time(0); err != nil {
+				i, err = indexOf("time")
+				if err != nil {
 					return
 				}
-
-				if anno.StartTime, err = v.Time(1); err != nil {
+				if anno.EndTime, err = v.Time(i); err != nil {
 					return
 				}
 
-				if anno.modTime, err = v.Int64(2); err != nil {
+				i, err = indexOf("start_time")
+				if err != nil {
+					return
+				}
+				if anno.StartTime, err = v.Time(i); err != nil {
 					return
 				}
 
-				if anno.Text, err = v.String(3); err != nil {
+				i, err = indexOf("modified_time_ns")
+				if err != nil {
+					return
+				}
+				if anno.modTime, err = v.Int64(i); err != nil {
 					return
 				}
 
-				if anno.Type, err = v.String(4); err != nil {
+				i, err = indexOf("text")
+				if err != nil {
+					return
+				}
+				if anno.Text, err = v.String(i); err != nil {
 					return
 				}
 
-				if anno.ID, err = v.String(5); err != nil {
+				i, err = indexOf("id")
+				if err != nil {
+					return
+				}
+				if anno.ID, err = v.String(i); err != nil {
 					return
 				}
 
+				anno.Tags = chronograf.AnnotationTags{}
+
+				// Collect all other columns as tags
+				for i, vv := range v {
+					key := s.Columns[i]
+					invalid := chronograf.ValidateAnnotationTagKey(key)
```

Comment:
*nit* It's common practice to always call returned errors `err`.

Reply:
hmm it will reassign the named return variable `err`. Then if the last `key` in the range was a reserved tag key, the entire function will return an err when it should not have.

I could

- reassign `err` to `nil` after the `range`
- remove the used of named `err` return
- leave as is

what do you think?

Reply:
if you do ` err :=` it should be in it's own scope and wont conflict with the returned error.